src/parser/core/modules/Interrupts.tsx

```diff
@@ -0,0 +1,143 @@
+
+interface SeverityTiers {
+	[key: number]: number
+}
+
+// used for timeline viewing by giving you a nice 30s window
+const TIMELINE_UPPER_MOD: number = 30000
+
+export abstract class Interrupts extends Module {
+	static handle: string = 'interrupts'
+	static title: MessageDescriptor = t('core.interrupts.title')`Interrupted Casts`
+
+	@dependency private timeline!: Timeline
+	@dependency private globalCooldown!: GlobalCooldown
+	@dependency private suggestions!: Suggestions
+	@dependency private data!: Data
+
+	private currentCast?: CastEvent | undefined
+	private droppedCasts?: CastEvent[] = []
+	private missedTimeMS: number = 0
+
+	/**
+	 * Implementing modules MAY override the icon to be used for the suggestion,
+	 * though, let's face it – interject is pretty much the perfect one.
+	 */
+	protected icon: string = ACTIONS.INTERJECT.icon
+
+	/**
+	 * Implementing modules MAY override the severity tiers for interrupted casts
+	 */
+	protected severity: SeverityTiers = {
+		2: SEVERITY.MEDIUM,
+		5: SEVERITY.MAJOR,
+	}
+
+	/**
+	 * Implementing modules MAY override the default suggestion text
+	 */
+	protected suggestionContent: JSX.Element | string = <Trans id="core.interrupts.suggestion.content">
+		Avoid interrupting casts by either prepositioning yourself or utilizing slidecasting where possible.
+	</Trans>
+
+	/**
+	 * Implementing modules MAY override this function to provide specific text if they wish for the 'why'
+	 * @param missedCasts
+	 * @param missedTime
+	 */
+	private suggestionWhy(missedCasts: CastEvent[], missedTime: number): JSX.Element | string {
+		return <Trans id="core.interrupts.suggestion.why">You missed { missedCasts.length } casts (approximately { this.parser.formatDuration(missedTime) } of total casting time) due to interruption.</Trans>
+	}
+
+	protected init() {
+		this.addEventHook('begincast', {by: 'player'}, this.onBeginCast)
+		this.addEventHook('cast', {by: 'player'}, this.onCast)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onBeginCast(event: CastEvent) {
+		// if they started casting something, then cancel and start something else,
+		// that's definitely an interrupted cast
+		if (this.currentCast) {
+			this.pushDropCasts(event)
+		}
+		this.currentCast = event
+	}
+
+	private onCast(event: CastEvent) {
+		const guid = event.ability.guid
+		// if the thing they started casting doesn't match up with what they cast, then
+		// that's an interrupted cast
+		if (this.currentCast && guid !== this.currentCast.ability.guid) {
+			this.pushDropCasts(event)
+		}
+		this.currentCast = undefined
+	}
+
+	private pushDropCasts(currentEvent: CastEvent) {
+		this.missedTimeMS += Math.min(this.globalCooldown.getEstimate(), currentEvent.timestamp - this.currentCast!.timestamp)
+		this.droppedCasts!.push(currentEvent)
+	}
+
+	private onComplete() {
+		this.suggestions.add(new TieredSuggestion({
+			icon: this.icon,
+			tiers: this.severity,
+			value: this.droppedCasts!.length,
```

Comment:
This can be straight up `this.droppedCasts.length` once you feex the optionalness never being used.